src/Mobs/Monster.h

```diff
@@ -194,7 +194,10 @@ class cMonster :
 	int FindFirstNonAirBlockPosition(double a_PosX, double a_PosZ);
 
 	/** Returns if the ultimate, final destination has been reached */
-	bool ReachedFinalDestination(void) { return ((m_FinalDestination - GetPosition()).SqrLength() < (m_AttackRange * m_AttackRange)); }
+	bool ReachedFinalDestination(void) { return ((m_FinalDestination - GetPosition()).Length() < GetWidth()); }
```

Comment:
For these length comparisons, it might be better to compare the square length to the width squared, because squaring is faster than square rooting. But thats just a micro optimization.